sycl/source/detail/helpers.cpp

```diff
@@ -23,6 +23,8 @@ std::vector<RT::PiEvent> getOrWaitEvents(std::vector<cl::sycl::event> DepEvents,
   std::vector<RT::PiEvent> Events;
   for (auto SyclEvent : DepEvents) {
     auto SyclEventImplPtr = detail::getSyclObjImpl(SyclEvent);
+    if (SyclEventImplPtr->getHandleRef() == nullptr)
+      continue;
```

Comment:
Is this only possible for default-constructed events? What will host events return here?

Reply:
correct, this is only possible for default constructed events.  It was changed because `is_host` now gets the context so it can answer the question of host or not.